# test_series_descriptions.csv

---

Nuestras carpetas están organizadas en un principio en "../data/raw/" y tenemos las carpetas:

 `test_images`, `train_images`,
 
 y los archivos:
 
  `sample_submission.csv`, `test_series_descriptions.csv`, `train_label_coordinates.csv`, `train_series_descriptions.csv` y `train.csv`.

---

### Step 1: Problem statement and data collection

**Analysis Objective**

**Context os the analysis**


In [7]:
# Import the dataset to start working with it

import pandas as pd 

df_test_series = pd.read_csv("../data/raw/test_series_descriptions.csv")
df_test_series.head()

study_id   series_id series_description
0  44036939  2828203845        Sagittal T1
1  44036939  3481971518           Axial T2
2  44036939  3844393089   Sagittal T2/STIR

#### **Files**

**test_series_descriptions.csv**

- `study_id` --> The study ID. Each study may include multiple series of images.

- `series_id` --> The imagery series ID

- `series_description` --> La orientación de la radiografía

    - >¿Cuáles son todas las orientaciones?

---

### Step 2: Exploration and data cleaning

- #### 2.1 Summarize DF

In [4]:
# Obtain dimensions

rows, columns = df_test_series.shape

print(f"The dimensions of this dataset are: {rows} Rows and {columns} Columns")

The dimensions of this dataset are: 3 Rows and 3 Columns


In [5]:
# Obtain information about data types and non-null values

df_test_series.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3 entries, 0 to 2
Data columns (total 3 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   study_id            3 non-null      int64 
 1   series_id           3 non-null      int64 
 2   series_description  3 non-null      object
dtypes: int64(2), object(1)
memory usage: 204.0+ bytes


In [6]:
# Check null values

null_var = df_test_series.isnull().sum().loc[lambda x: x > 0]

num_of_null_var = len(null_var)

print(f"{null_var}\n\nThe number of null variables are {num_of_null_var}")

Series([], dtype: int64)

The number of null variables are 0


---------**Conclusiones**---------

- There are a total of 3 rows (Severity Scores) y 3 columns.

- Podemos ver que `study_id` es el mismo en las tres filas '44036939' (es la misma persona). Cambia `series_id`(la radiografía) y `series_description` (nos dice la orientación de la radiografía). Como conclusión podemos sacar que es posible que hayan varias radiografías de la misma persona. ¿A que nos lleva esto?¿Qué nos debemos de preguntar?

- The data has 1 categorical characteristics (series_description) and 2 numerical.

---

- #### 2.2 Eliminate duplicates

In [8]:
# Obtain the number of duplicate values in our dataset

print(f"The number of duplicate values in the dataset is {df_test_series.duplicated().sum()}")

The number of duplicate values in the dataset is 0
